webview/src/plots/components/ZoomablePlot.tsx

```diff
@@ -1,42 +1,35 @@
 import React, { useEffect, useRef } from 'react'
-import VegaLite, { VegaLiteProps } from 'react-vega/lib/VegaLite'
+import { useDispatch } from 'react-redux'
+import VegaLite from 'react-vega/lib/VegaLite'
+import { setZoomedInPlot } from './webviewSlice'
 import styles from './styles.module.scss'
 import { GripIcon } from '../../shared/components/dragDrop/GripIcon'
 
-export interface ZoomablePlotProps {
-  renderZoomedInPlot: (
-    plot: VegaLiteProps,
-    id: string,
-    refresh?: boolean
-  ) => void
-}
-
-interface ZoomablePlotOwnProps extends ZoomablePlotProps {
-  plotProps: VegaLiteProps
+interface ZoomablePlotProps {
+  plotProps: string
```

Comment:
I might revisit this later, but this is the fix I found to make sure all plots get rendered. Otherwise, Vega complains that it cannot set value to read only property id (I'm paraphrasing) or group...

We lose type validation that's the downside, but other than getting all plots rendered, another plus side is that passing strings instead of objects is that if a parent re-renders, the string will stay the same while an object might be passed with a new reference.

Reply:
This is the main cause of performance drop. Like I said before, this fixed the problem, but not definitely not at the root. I thought we could leave it as is for now, but that is definitely not an option now with results like this. `JSON.parse` can get quite expensive, who knew? I think I might have found the real solution for the original problem so that we can get rid of stringify'ed props.